Emby.Server.Implementations/Library/Validators/CollectionPostScanTask.cs

```diff
@@ -0,0 +1,155 @@
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>Task.</returns>
+        public async Task Run(IProgress<double> progress, CancellationToken cancellationToken)
+        {
+
+            var movies = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                IncludeItemTypes = new[] { nameof(Movie) },
+                IsVirtualItem = false,
+                OrderBy = new List<ValueTuple<string, SortOrder>>
+                {
+                new ValueTuple<string, SortOrder>(ItemSortBy.SortName, SortOrder.Ascending)
+                },
+                Recursive = true
+            });
+
+            var boxSets = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                IncludeItemTypes = new[] { nameof(BoxSet) },
+                CollapseBoxSetItems = false,
+                Recursive = true
+            });
+
+            var numComplete = 0;
+            var count = movies.Count;
+
+            var collectionNameMoviesMap = new Dictionary<string, List<Movie>>();
+            foreach (var m in movies)
+            {
+                var movie = m as Movie;
+                if (movie != null && movie.CollectionName != null)
+                {
+                    var movieList = new List<Movie>();
+                    if (collectionNameMoviesMap.TryGetValue(movie.CollectionName, out movieList))
+                    {
+                        if (!movieList.Any(m => m.Id == movie.Id))
+                        {
+                            movieList.Add(movie);
+                            collectionNameMoviesMap[movie.CollectionName] = movieList;
+                        }
+                    }
+                    else
+                    {
+                        collectionNameMoviesMap[movie.CollectionName] = new List<Movie> { movie };
+                    }
+
+                }
+
+                numComplete++;
+                double percent = numComplete;
+                percent /= count * 2;
+                percent *= 100;
+
+                progress.Report(percent);
+            }
+
+            foreach (var pair in collectionNameMoviesMap)
+            {
+                try
+                {
+                    var collectionName = pair.Key;
+                    var movieList = pair.Value;
+
+                    var boxSet = boxSets.FirstOrDefault(b => b != null ? b.Name == collectionName : false) as BoxSet;
+                    if (boxSet == null)
+                    {
+                        // won't automatically create collection if only one movie in it
+                        if (movieList.Count >= 2)
+                        {
+                            boxSet = await _collectionManager.CreateCollectionAsync(new CollectionCreationOptions
+                            {
+                                Name = collectionName,
+                                IsLocked = true
```

Comment:
Why locked?

Reply:
`IsLocked = true` is also defualt when create collection manually. If you create a collection with `IsLocked = false` and the collection don't have correspond online data, then the collection will don't have cover/image. On the other hand, when set `IsLocked = true`, jellyfin will auto generate a cover/image based on local file/video. I think it's more friendly and beautiful.

Reply:
Alright